### ISSUE TITLE
Implement memory-efficient trustworthiness metric in Rust

### ISSUE THREAD

**Issue by Trecek** [TalonT-Org/spectral-init#212]
# Implement memory-efficient trustworthiness metric in Rust

## Background

Our MERFISH benchmark pipeline (`tests/visual_eval/generate_merfish_comparisons.py`) calls `sklearn.manifold.trustworthiness()` to evaluate embedding quality. This metric measures whether k-nearest neighbors in the high-dimensional space remain close in the 2D embedding.

The sklearn implementation is O(n²) in memory — it materializes three full `(n, n)` dense arrays simultaneously:
1. `pairwise_distances(X)` → `(n, n) float64`
2. `np.argsort(dist, axis=1)` → `(n, n) int64`
3. `np.zeros((n, n), dtype=int)` → inverted rank index

Measured impact:
- 10K cells: ~2.4 GB from trustworthiness alone
- 20K cells: ~9.6 GB
- 100K cells: ~240 GB (infeasible)

The actual UMAP pipeline (preprocessing, graph build, spectral init, SGD) only needs ~3.5 GB at 100K. This single evaluation metric is the sole blocker for scaling the benchmark to 100K (issue #210).

## What trustworthiness computes

For each point `i`, trustworthiness checks whether its `k` nearest neighbors in the original high-dimensional space `X` are also among its `k` nearest neighbors in the embedding `Y`. Points that are far in `X` but close in `Y` are penalized. The formula:

```
T(k) = 1 - (2 / (n * k * (2n - 3k - 1))) * Σ_i Σ_j (r(i,j) - k)
```

where `r(i,j)` is the rank of point `j` in the distance ordering from point `i` in `X`, and the inner sum is over points `j` that are in the `k`-neighborhood of `i` in `Y` but NOT in the `k`-neighborhood of `i` in `X`.

Reference: Venna & Kaski (2006), "Local multidimensional scaling". sklearn source: `sklearn/manifold/_t_sne.py` function `trustworthiness`.

## Requirements

1. **Implement `trustworthiness(X, Y, k)` in Rust** as a utility in this crate (likely `src/metrics/` or `src/utils/`). It should accept:
   - `X`: high-dimensional data, `(n, d)` ndarray or slice
   - `Y`: 2D embedding, `(n, 2)` ndarray or slice
   - `k`: neighborhood size (typically 15)
   - Return: `f64` in `[0, 1]`

2. **Memory must be O(n·k), not O(n²).** The key insight: you never need the full pairwise distance matrix. For each point you only need its `k` nearest neighbors in both spaces. Use a k-NN approach:
   - Build a k-NN index on `X` (e.g., brute-force with a partial sort, or a ball tree) to get each point's `k` neighbors in high-dim
   - Build a k-NN index on `Y` to get each point's `k` neighbors in the embedding
   - Compute the trustworthiness penalty from the set differences
   - Peak memory: two `(n, k)` index arrays + two `(n, k)` distance arrays = `O(n·k)`

3. **Expose via FFI or as a Python-callable binary** so the benchmark scripts can call it instead of sklearn's version. Options:
   - Write results to a file that Python reads
   - PyO3 bindings (if scope permits)
   - CLI binary that takes .npy paths and prints the score

4. **Validate against sklearn** at 10K where both implementations can run. Results should match to within floating-point tolerance (±1e-6).

## Relevant context

- Project architecture: see `CLAUDE.md` and `src/` layout
- The crate already uses `ndarray` for dense arrays and `sprs` for sparse matrices
- `faer` is available for linear algebra
- The existing benchmark data lives at `tests/visual_eval/merfish_data/` (10K, committed) and `temp/merfish_100k/` (100K, runtime-generated)
- The metric is currently called in `generate_merfish_comparisons.py` lines ~200-210 (baseline phase) and ~350-370 (compare phase)
- `knn_preservation` is a related metric in the same file that also computes k-NN — there may be opportunity to share the neighbor lookup
- This is a blocker for issue #210 (Scale MERFISH benchmark to 100K cells)

## Out of scope

- Rewriting other metrics (silhouette, Moran's I, SNA, etc.)
- Changing the UMAP pipeline itself
- Modifying the spectral init algorithm
